### PR TITLE
Add nested array test

### PR DESCRIPTION
**Pull request by vicsn**
## Motivation

Adds a test for maximally nested arrays of maximum size.
Also in the previous commit tested a related program with as many arrays of maximum size as possible.

No wild memory usage or runtime was observed.

### CODE REVIEW

File: synthesizer/src/vm/mod.rs [hunk 1]
```diff
@@ -3448,4 +3448,232 @@ function adder:
...
+                None,
+                rng,
+            )
+            .unwrap();
+
+        // Verify the execution
+        vm.check_transaction(&transaction, None, rng).unwrap();
+
+        println!(
+            "Successfully deployed and executed program with array nesting depth {} and {} elements per dimension",
```
Comment (d0cd) [inline synthesizer/src/vm/mod.rs:3609]: Overall looks great! A couple points:
- It could be helpful to assert the max depth here just so we "fix" the actual size that was tested.
- It could be helpful to test in a finalize context and assert the cost.
- Also as a standard pattern, we should assert that the execution transactions were in fact accepted.

**Review by d0cd — APPROVED** [review]
Thanks for the test! Left some minor feedback, feel free to either address or merge the PR and I'm happy to make tweaks in `test/ecdsa-bytes`

File: synthesizer/src/vm/mod.rs (hunk 1 above)
Comment (vicsn) [reply synthesizer/src/vm/mod.rs:3609]: Great suggestions.

e7ce898e1

Only thing I didn't do is assert the cost, feels out of scope
